### PR TITLE
chore(contentful): export data from contentful

### LINKED ISSUE #10465: contentful export as an alternative to using sync api
More info in the description of this pull request #10432

### PR DESCRIPTION
**Pull request by busla**
closes #10465 

- chore: added contentful exporter config
- fix: added content-export sh script

## What

This is an attempt to remove the need for the indexer to walk the whole contentful model tree when a parent is added/changed/deleted to collect all the data before reindexing. That takes between 20-30 minutes because of numerous http requests to contentful while walking down the tree.

Using data export (well only data needed for elastic in this case) takes 2 minutes.

The exported dump is then copied to s3 for the search-indexer to get and remap, preferably using some json-stream lib to not have to load everything to memory when starting the iterator.

The idea is to create an alias that points the current index, remap the dump and restore it to a new index and then move the alias to the new one.

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by krrrrrrrrrrr — APPROVED** [review]
LGTM

**Review by brynjarorng — CHANGES_REQUESTED** [review]

File: scripts/config/contentful-config.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "content_type=menu",
+    "content_type=groupedMenu",
+    "content_type=organizationPage",
+    "content_type=organizationSubpage",
+    "content_type=projectPage",
+    "content_type=supportQNA",
+    "content_type=link",
+    "content_type=frontpage",
+    "content_type=enhancedAsset"
+  ],
```
[the remark this answers is not in the document]
Comment (busla) [reply scripts/config/contentful-config.json:23]: these are the only content-types needed for elasticsearch.

the idea is to export only the data that is needed to not have to preprocess data that will be discarded anyway.

but you are right, we should try to pull this object from the search-indexer app, good catch.

File: scripts/contentful-export.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+s3_bucket="${env}-island-is-contentful-data-export"
+
+function get_param() {
+  local name=$1
+  aws ssm get-parameter --name "$name" --with-decryption | jq -r '.Parameter.Value'
+}
+
+function contentful_export() {
+# FIXME: incorrect space-id in ssm
+# --space-id "$(get_param "/k8s/api-catalogue/CONTENTFUL_SPACE_ID")"
```
[the remark this answers is not in the document]
Comment (busla) [reply scripts/contentful-export.sh:19]: I just got clarification on why it was incorrect, will update, thanks.

File: scripts/config/contentful-config.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (eirikurn) [reply scripts/config/contentful-config.json:23]: Hm. Doesn't the indexer also query/include all linked entries from these root entries? I assume the exporter doesn't do that.

Why not just export/backup everything?

**Review by eirikurn — COMMENTED** [review]
Please add some context to the PR so I know what I'm reviewing. What is the purpose of this?

**Comment by busla** [comment]
this PR should be a draft still, will change.

PR description updated, thanks.

File: scripts/config/contentful-config.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (busla) [reply scripts/config/contentful-config.json:23]: Just added description to the PR for better context.

According to @RunarVestmann these are the only content types remaped and added to the elastic index.

We could dump everything but why bother when we know what data will end up eventually in elastic?

[the remark this answers is not in the document]
Comment (busla) [reply scripts/config/contentful-config.json:23]: The `queryEntries` list is copied from [here](https://github.com/island-is/island.is/blob/main/libs/cms/src/lib/environments/environment.ts#L5) but should be pulled in.

**Pull request closed without merging**